crates/solver-service/src/server.rs

```diff
@@ -507,3 +538,112 @@ async fn forward_to_discovery_service(
 		},
 	}
 }
+
+/// Calculates the profit margin percentage for an order.
+///
+/// The profit margin is calculated as:
+/// Profit = Total Input Amount - Total Output Amount - Execution Costs
```

Comment:
This would only work if we're swapping the same value tokens (USDC -> USDC). We'll need to involve solver-pricing to get USD equivalent of a token (e.g. ETH -> USDC) we need get ETH/USDC pair token price and THEN compare normalised amounts.